src/Uno.UI/Extensions/UIElementExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Windows.UI.Xaml;
+using Uno.Extensions;
+using Uno.Logging;
+
+namespace Uno.UI.Extensions
+{
+	public static partial class UIElementExtensions
+	{
+		public static Thickness GetPadding(this UIElement uiElement)
+		{
+			if(uiElement is FrameworkElement fe && fe.TryGetPadding(out var padding))
+			{
+				return padding;
+			}
+
+			var property = uiElement.GetDependencyPropertyUsingReflection("PaddingProperty");
+			return property == null ? Thickness.Empty : (Thickness)uiElement.GetValue(property);
+		}
+
+		public static Thickness GetBorderThickness(this UIElement uiElement)
+		{
+			if (uiElement is FrameworkElement fe && fe.TryGetBorderThickness(out var borderThickness))
+			{
+				return borderThickness;
+			}
+
+			var property = uiElement.GetDependencyPropertyUsingReflection("BorderThicknessProperty");
+			return property == null ? Thickness.Empty : (Thickness)uiElement.GetValue(property);
```

Comment:
This must be a safe-cast
```suggestion
			return property != null && uiElement.GetValue(property) is Thickness t ? t : default;
```